src/structure/real.rs

```diff
@@ -0,0 +1,267 @@
+use num::{Float, Num, FromPrimitive};
+use std::ops::{Neg, AddAssign, MulAssign, SubAssign, DivAssign};
+use cmp::ApproxEq;
+use structure::FieldApprox;
+
+#[allow(missing_docs)]
+
+/// Trait shared by all approximate reals.
+///
+/// Approximate reals are approximate fields aquipped with funtinos that are commonly used on
+/// reals. The results of those functions only have to be approximately equal to the actual
+/// theoretical values.
+pub trait RealApprox: Copy + Num + FieldApprox + FromPrimitive +
+                      Neg<Output = Self> + AddAssign + MulAssign + SubAssign + DivAssign +
+                      ApproxEq<Eps = Self> + Ord {
+    fn floor(self) -> Self;
+    fn ceil(self) -> Self;
+    fn round(self) -> Self;
+    fn trunc(self) -> Self;
+    fn fract(self) -> Self;
+    fn abs(self) -> Self;
+    fn signum(self) -> Self;
+    fn is_sign_positive(self) -> bool;
+    fn is_sign_negative(self) -> bool;
+    fn mul_add(self, a: Self, b: Self) -> Self;
+    fn recip(self) -> Self;
+    fn powi(self, n: i32) -> Self;
+    fn powf(self, n: Self) -> Self;
+    fn sqrt(self) -> Self;
+    fn exp(self) -> Self;
+    fn exp2(self) -> Self;
+    fn ln(self) -> Self;
+    fn log(self, base: Self) -> Self;
+    fn log2(self) -> Self;
+    fn log10(self) -> Self;
+    fn max(self, other: Self) -> Self;
+    fn min(self, other: Self) -> Self;
+    fn abs_sub(self, other: Self) -> Self;
+    fn cbrt(self) -> Self;
+    fn hypot(self, other: Self) -> Self;
+    fn sin(self) -> Self;
+    fn cos(self) -> Self;
+    fn tan(self) -> Self;
+    fn asin(self) -> Self;
+    fn acos(self) -> Self;
+    fn atan(self) -> Self;
+    fn atan2(self, other: Self) -> Self;
+    fn sin_cos(self) -> (Self, Self);
+    fn exp_m1(self) -> Self;
+    fn ln_1p(self) -> Self;
+    fn sinh(self) -> Self;
+    fn cosh(self) -> Self;
+    fn tanh(self) -> Self;
+    fn asinh(self) -> Self;
+    fn acosh(self) -> Self;
+    fn atanh(self) -> Self;
+}
```

Comment:
I think this would be needlessly confusing because this would increase the number of traits that do not exactly match a specific abstract mathematical concept.